src/WebJobs.Extensions.DurableTask/EntityScheduler/Entity.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Statically accessible context for entity operations.
+    /// </summary>
+    public static class Entity
+    {
+        private static readonly AsyncLocal<IDurableEntityContext> EntityContext
+            = new AsyncLocal<IDurableEntityContext>();
+
+        /// <summary>
+        /// The context of the currently executing entity.
+        /// </summary>
+        public static IDurableEntityContext Context => EntityContext.Value;
+
+        /// <summary>
+        /// The key of the currently executing entity.
+        /// </summary>
+        public static string Key => EntityContext.Value.Key;
+
+        /// <summary>
+        /// The entity reference for the currently executing entity.
+        /// </summary>
+        public static EntityId Self => EntityContext.Value.Self;
```

Comment:
I know I tend to be picky about names, but I do wonder if `Self` is a little too ambiguous. In languages like Python, `self` is essentially the same thing as the `this` pointer in C-style languages. For that reason, it confuses me when something named `Self` refers to an `EntityId` rather than an `IDurableEntityContext` or even the current entity class instance (e.g. `ChatRoom`). I feel it might be more obvious what the type of the property is if it were instead simply named `Id`.

Reply:
I have removed this field from this class.

Are you making the case for changing the property`Self` in `IDurableEntityContext` as well? It would be a breaking change.

Reply:
I would prefer we update `IDurableEntityContext` as well. I realize it is a breaking change, though I assume it's not commonly used yet. My hope is that we can have most of the breaking changes done by the next release (more breaking changes in other areas are already coming).

Reply:
Sure, I agree. I will put the rename in a separate PR.